From d2cb323a270c5188d6f316013db14c94eeff3603 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Wed, 19 Feb 2025 14:26:10 +0100
Subject: [PATCH 1594/1695] dts: stm32: drop unsupported interrupt bindings for
 PWR IRQ

With the code ported to the generic interrupt framework, drop the
old properties wakeup-parent and st,wakeup-pin-number.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f46cf122b035d6c714cd257ecccfa7ace4e366f2 dts: stm32: drop unsupported interrupt bindings for PWR IRQ]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: Iefc1c3c0ce12d2d46f34c654ed72cf8f7e952c81
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434938
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp135f-dk.dts  | 6 ------
 core/arch/arm/dts/stm32mp215f-dk.dts  | 2 --
 core/arch/arm/dts/stm32mp235f-dk.dts  | 2 --
 core/arch/arm/dts/stm32mp257f-dk.dts  | 2 --
 core/arch/arm/dts/stm32mp257f-ev1.dts | 2 --
 5 files changed, 14 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp135f-dk.dts b/core/arch/arm/dts/stm32mp135f-dk.dts
index d394c4cd4..d9542e48b 100644
--- a/core/arch/arm/dts/stm32mp135f-dk.dts
+++ b/core/arch/arm/dts/stm32mp135f-dk.dts
@@ -35,8 +35,6 @@
 
 	magic_wol: magic_wol {
 		compatible = "st,stm32mp1-pwr-irq-user";
-		wakeup-parent = <&pwr_irq>;
-		st,wakeup-pin-number = <2>;
 		interrupts-extended = <&pwr_irq 1 IRQ_TYPE_EDGE_FALLING>;
 		wakeup-source;
 		status = "okay";
@@ -70,8 +68,6 @@
 
 	typec_wakeup: typec_wakeup {
 		compatible = "st,stm32mp1-pwr-irq-user";
-		wakeup-parent = <&pwr_irq>;
-		st,wakeup-pin-number = <5>;
 		interrupts-extended = <&pwr_irq 4 IRQ_TYPE_EDGE_FALLING>;
 		wakeup-source;
 		st,notif-it-id = <1>;
@@ -211,8 +207,6 @@
 		compatible = "st,stpmic1";
 		reg = <0x33>;
 		status = "okay";
-		wakeup-parent = <&pwr_irq>;
-		st,wakeup-pin-number = <1>;
 		interrupts-extended = <&pwr_irq 0 IRQ_TYPE_EDGE_FALLING>;
 		wakeup-source;
 		st,pmic-it-id = <IT_PONKEY_F IT_PONKEY_R>;
diff --git a/core/arch/arm/dts/stm32mp215f-dk.dts b/core/arch/arm/dts/stm32mp215f-dk.dts
index 0969d1973..d1994f5ed 100644
--- a/core/arch/arm/dts/stm32mp215f-dk.dts
+++ b/core/arch/arm/dts/stm32mp215f-dk.dts
@@ -70,11 +70,9 @@
 		compatible = "st,stpmic2l";
 		reg = <0x33>;
 		interrupts-extended = <&pwr 0 IRQ_TYPE_EDGE_FALLING>;
-		st,wakeup-pin-number = <1>;
 		wakeup-source;
 		st,pmic-it-id = <IT_PKEY_FA IT_PKEY_RI>;
 		st,notif-it-id = <0 1>;
-		wakeup-parent = <&pwr>;
 		status = "okay";
 
 		regulators {
diff --git a/core/arch/arm/dts/stm32mp235f-dk.dts b/core/arch/arm/dts/stm32mp235f-dk.dts
index 16a8a0822..1bcc40e0d 100644
--- a/core/arch/arm/dts/stm32mp235f-dk.dts
+++ b/core/arch/arm/dts/stm32mp235f-dk.dts
@@ -73,11 +73,9 @@
 		compatible = "st,stpmic2";
 		reg = <0x33>;
 		interrupts-extended = <&pwr 0 IRQ_TYPE_EDGE_FALLING>;
-		st,wakeup-pin-number = <1>;
 		wakeup-source;
 		st,pmic-it-id = <IT_PKEY_FA IT_PKEY_RI>;
 		st,notif-it-id = <0 1>;
-		wakeup-parent = <&pwr>;
 		status = "okay";
 
 		regulators {
diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index d698edf28..a6cc2c5f1 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -73,11 +73,9 @@
 		compatible = "st,stpmic2";
 		reg = <0x33>;
 		interrupts-extended = <&pwr 0 IRQ_TYPE_EDGE_FALLING>;
-		st,wakeup-pin-number = <1>;
 		wakeup-source;
 		st,pmic-it-id = <IT_PKEY_FA IT_PKEY_RI>;
 		st,notif-it-id = <0 1>;
-		wakeup-parent = <&pwr>;
 		status = "okay";
 
 		regulators {
diff --git a/core/arch/arm/dts/stm32mp257f-ev1.dts b/core/arch/arm/dts/stm32mp257f-ev1.dts
index 548a3e137..a2e405565 100644
--- a/core/arch/arm/dts/stm32mp257f-ev1.dts
+++ b/core/arch/arm/dts/stm32mp257f-ev1.dts
@@ -73,11 +73,9 @@
 		compatible = "st,stpmic2";
 		reg = <0x33>;
 		interrupts-extended = <&pwr 0 IRQ_TYPE_EDGE_FALLING>;
-		st,wakeup-pin-number = <1>;
 		wakeup-source;
 		st,pmic-it-id = <IT_PKEY_FA IT_PKEY_RI>;
 		st,notif-it-id = <0 1>;
-		wakeup-parent = <&pwr>;
 		status = "okay";
 
 		regulators {
-- 
2.47.2

